src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,150 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin("loogiin");
+        user.setPassword("password");
+        user.setAge(20);
+    }
+
+    @Test
+    void register_validUser_ok() {
+        assertEquals(registrationService.register(user), user);
+        assertEquals(Storage.people.size(), 1);
+    }
+
+    @Test
+    void register_login_notOk() {

```

Comment:
Best practice it to isolate test methods, create one method for particular test case. 
If you want to test several conditions or options for one method, you can use `@ParameterizedTest`.
When your test 'register_login_notOk' fails, you won't know which condition is not working.

Reply:
Why don't we know? There we can see which assert fails and row number:
![image](https://github.com/mate-academy/jv-registration-validation/assets/71705897/4c869373-eb1b-406c-a60a-194d94c3e27c)